workflow/controller/operator.go

```diff
@@ -276,6 +266,19 @@ func (woc *wfOperationCtx) operate(ctx context.Context) {
 		woc.computeMetrics(woc.execWf.Spec.Metrics.Prometheus, localScope, realTimeScope, true)
 	}
 
+	// Reconciliation of Outputs (Artifacts). See reportOutputs() of executor.go
+	woc.taskResultReconciliation()
```

Comment:
I do see this will now get executed in the case of `woc.wf.Status.Phase == wfv1.WorkflowUnknown` which before it wouldn't. This may be right but want to make sure. I also like not having the `if` statement if it's unnecessary. I believe this would happen for a just submitted workflow, or it appears a [resubmitted](https://github.com/argoproj/argo-workflows/blob/master/workflow/util/util.go#L752) workflow. It does seem like this probably won't ever be harmful - if there's a WorkflowTaskResult of the same name then we probably ought to use it, right?

Reply:
(unless there's some weird case that a previous Workflow of the same name still had a WorkflowTaskResult lying around from before...but I believe a WorkflowTaskResult should get deleted at the same time as the Workflow it relates to)

Reply:
@juliev0 if I'm not mistaken, this change resolved some intermittent test failures I was receiving. I believe the issue was that artifacts could be created and garbage collected while `wfv1.WorkflowUnknown` was set. This would cause `taskReconciliation` to not run and garbage collection to fail since the node status wouldn't get updated with the artifact info.

Reply:
huh - that's weird that artifacts were created and garbage collected while `wfv1.WorkflowUnknown` was set. I wonder if there could be a case of `wfv1.WorkflowUnknown` that I'm unaware of if so. Anyway, it seems to be harmless.

Reply:
Agreed. I'm not 100% on this, but as you say the change should be harmless.

Reply:
Quick update here: irrespective of if artifacts can be written when `wfv1.WorkflowUnknown` is set, the more important thing with this change is that the reconciliation runs even when the workflow is marked as complete as per `woc.wf.Labels[common.LabelKeyCompleted] == "true"`